### ISSUE TITLE
Speed limit

### ISSUE THREAD

**Issue by Franzrlando** [andrey-pr/OpenBafangTool#21]
Hi, i have to say that u are doing a amazing job!
but i have a question, there is no way to used to change the speed limit?
i cant do it slower or faster the programm changes it automatically to 25km

**Comment by andrey-pr** [comment]
> Do you mean that the program changed the limit to 25km/h, and now you can't put it back to how it was before?

No. If limit is bigger than 25kmh program will not change it at all. Only if limit is 25kmh on start, you can set it lower or back to 25. On some firmwares its impossible to change limit at all

**Comment by andrey-pr** [comment]
> Hi, i have to say that u are doing a amazing job!
> but i have a question, there is no way to used to change the speed limit?
> i cant do it slower or faster the programm changes it automatically to 25km

Probably you have firmware with locked limit

**Comment by Franzrlando** [comment]
hi, yes the controller comes with 25km/h from stock, but in the besst app from bafang is possible to change the speed, but in your app i cant change it (lower or higher) the app set it again on 25km/h same happen with the circunference of the wheel and wheel size

**Comment by ncdm-stldr** [comment]
I have a friend that encountered the same issue i.e. when changing the value, it returns back to 25, even before clicking write or doing anything.
I am surprised because I did the same with a bike of the same model in the past and I had no issue. However, I used a earlier version of OpenBafangTool (it was https://github.com/andrey-pr/OpenBafangTool/releases/tag/2.0.1-rc).

I wonder what could cause the UI to change even before clicking on write? It doesn't happen in demo mode.
